tough-kms/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! tough-kms implements the `KeySource` trait found in [tough, a Rust TUF client](https://github.com/awslabs/tough).
+//!
+//! By implementing this trait, AWS KMS can become a source of keys used to sign a [TUF repository](https://theupdateframework.github.io/).
+//!
+//! # Testing
+//!
+//! Unit tests are run in the usual manner: `cargo test`.
+
+#![forbid(missing_debug_implementations, missing_copy_implementations)]
+#![deny(rust_2018_idioms)]
+// missing_docs is on its own line to make it easy to comment out when making changes.
+#![deny(missing_docs)]
+#![warn(clippy::pedantic)]
+#![allow(
+    clippy::module_name_repetitions,
+    clippy::must_use_candidate,
+    clippy::missing_errors_doc
+)]
+
+mod client;
+pub mod error;
+use ring::digest::{digest, SHA256};
+use ring::rand::SecureRandom;
+use rusoto_kms::{Kms, KmsClient, SignRequest};
+use snafu::{OptionExt, ResultExt};
+use std::collections::HashMap;
+use std::fmt;
+use tough::key_source::KeySource;
+use tough::schema::decoded::{Decoded, RsaPem};
+use tough::schema::key::{Key, RsaKey, RsaScheme};
+use tough::sign::Sign;
+
+/// Represents a Signing Algorithms for AWS KMS.
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub enum KmsSigningAlgorithm {
+    /// Signing Algorithm `RSASSA_PSS_SHA_256`
+    RsassaPssSha256,
+}
+
+impl Default for KmsSigningAlgorithm {
+    fn default() -> Self {
+        KmsSigningAlgorithm::RsassaPssSha256
+    }
+}
```

Comment:
We shouldn't do this; we want the user of the key source to always be aware of what key signing algorithm is in use, and a default implementation makes less sense when we eventually add ECDSA support to tough-kms.

For now, tuftool can hardcode the `RsassaPssSha256` variant; we'll figure out a solution there for ECDSA support.

```suggestion
```

Reply:
Done.